cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramRunStatusMonitorService.java

```diff
@@ -147,14 +156,18 @@ Set<ProgramRunId> terminatePrograms() {
         break;
       }
       for (RunRecordDetail record : stoppingRuns.values()) {
+        String provisionerName = SystemArguments.getProfileProvisioner(record.getSystemArgs());
+        ProgramRunId programRunId = record.getProgramRunId();
         // If this run was initiated by this CDAP instance to run on another instance, it will use the tethering
         // provisioner. The instance that actually runs it on the other side will have a non-null peer name.
-        String provisionerName = SystemArguments.getProfileProvisioner(record.getSystemArgs());
         if (record.getPeerName() == null && TetheringProvisioner.TETHERING_NAME.equals(provisionerName)) {
-          continue;
+          LOG.info("Forcing the termination of tethered program run {} as it should have stopped at {} ",
+                   programRunId, record.getTerminateTs());
+          programScannedForTermination.add(programRunId);
+          programStateWriter.killed(programRunId);
+          emitForceTerminatedRunsMetric(programRunId, record);
```

Comment:
shouldn't we have a continue here?

Reply:
oops, you're right! thanks for point that out.

Reply:
fixed